### PR TITLE
Reworks cult-EMPs affecting cult-IPCs

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do
This PR alters EMP code to check if the source is from the cult, if so it will not affect any IPCs that are also cult members. I also gave this process a visual clue, the cult shield sprite will show in the moment of the pulse as well as the cult EMPs having different sprites. Non cult IPCs will still be harmed, and the normal EMP sprites remain the same for all other EMP sources other than cult ones.

## Why It's Good For The Game
Having cult members unknowingly or mistakenly murder their fellow cultists is bad. This also gives IPCs more cult gameplay possibilities. Were before EMP runes and talismans couldnt be used by IPCs and had to be used away from IPCs by their fellow cultists, now they can be actively used by and around IPCs.

## Images of changes
g2ap5aB07m
<details>
  <summary>click this to see how old cult EMPs look (the same as normal EMPs)</summary>
  
wTYxH33wlu
  
</details>

## Changelog
:cl:
add: Adds cult unique emp effect sprites.
tweak: Tweaks how EMPs interact with cult IPCs when cult sourced.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/cult/talisman.dm [hunk 1]
```diff
@@ -231,7 +231,7 @@
 	. = ..()
 	user.visible_message("<span class='warning'>[user]'s hand flashes a bright blue!</span>", \
 						 "<span class='cultitalic'>You speak the words of the talisman, emitting an EMP blast.</span>")
-	empulse(src, 4, 8, 1)
+	empulse(src, 4, 8, 1,, TRUE)
```
Comment (trololiver112) [inline code/game/gamemodes/cult/talisman.dm:234]: What happened to the double ,, here?

Comment (kugamo) [reply code/game/gamemodes/cult/talisman.dm:234]: `/proc/empulse(turf/epicenter, heavy_range, light_range, log=0, cause = null, srccult = FALSE)` empulse calls for 5 arguments normally, I didnt want to supply information in the wrong slot.. if you understand what I mean. I want to pass in TRUE for the 6th argument, not the 5th. So I added another comma to make TRUE pass into the 6th.

Comment (kugamo) [reply code/game/gamemodes/cult/talisman.dm:234]: /proc/empulse(turf/epicenter, heavy_range, light_range, log=0, cause = null, srccult = FALSE) empulse calls for 5 arguments normally, I didnt want to supply information in the wrong slot.. if you understand what I mean. I want to pass in TRUE for the 6th argument, not the 5th. So I added another comma to make TRUE pass into the 6th.

File: code/game/objects/empulse.dm [hunk 2]
```diff
@@ -18,15 +18,19 @@
 		M << 'sound/effects/empulse.ogg'
 	for(var/atom/T in range(light_range, epicenter))
 		var/distance = get_dist(epicenter, T)
-		if(distance < 0)
-			distance = 0
-		if(distance < heavy_range)
-			T.emp_act(1)
-		else if(distance == heavy_range)
-			if(prob(50))
+		if(iscultist(T) && srccult == TRUE && ismachineperson(T))
```
Comment (AffectedArc07) [inline code/game/objects/empulse.dm:21]: you dont need `srccult == TRUE`, just `srccult`

**Comment by AffectedArc07** [comment]
I echo kyets concerns, hence why I tagged this as `Balance`. This isnt really a fix, and the way its done is kind of hacky, making a modification to the baseline EMP code to add a special "is cult" check to each EMP.

**Comment by trololiver112** [comment]
Cult rework was closed and is not being worked on by anyone at the moment i believe

**Comment by trololiver112** [comment]
Adding on to that previous. IPC wizards get to choose their species. If they choose IPC they are basically going hard mode. Cultists don’t have a say in their own conversion. And if they are in a small base where they just go converted and accidentally killed by an ally it just kinda sucks. Not to mention the talisman and disable tech are two separate sources for emp, so the difference is okay probably.

**Comment by kugamo** [comment]
> 
> 
> This is a balance change / feature, not really a fix.
> While cult is pretty weak right now, that won't be the case if cult rework is merged. So I'm not sure they need the buff.
> We also allow IPC wizards to kill themselves with the "disable tech" spell, so its weird we would not allow cultists to kill each other with the EMP spell.

I changed the title from Fix to Rework. And as far as the code quality goes, I am at the mercy of all yours' judgement. I think its a feature/change that would benefit the game even if the cult is being reworked. Having cult EMP kill Cult IPCs is a needless crutch that only serves to hinder the goals of the cult. ~~like the punishment for using the wrong final rune~~ 

> 
> 
> Adding on to that previous. IPC wizards get to choose their species. If they choose IPC they are basically going hard mode. Cultists don’t have a say in their own conversion. And if they are in a small base where they just go converted and accidentally killed by an ally it just kinda sucks. Not to mention the talisman and disable tech are two separate sources for emp, so the difference is okay probably.

This PR does not change the normal EMP nor its affects, so the wizard spell is fine. I agree on the point about IPCs' lack of choice in being cult or not in some cases, which in my opinion exemplifies the issue.

**Comment by TDSSS** [comment]
Imo this is a good change. IPC already are very weak, they shouldn't also cripple the cult's ability to use EMPs after they are converted.

**Comment by TDSSS** [comment]
@Kyep I think you're drawing some unfair comparisons, there.

> Traitors who use EMP implants kill IPC traitors, including themselves.

Traitors are not a team antag, and if you kill yourself you have nobody to blame but yourself.

> Changelings who use EMP kill friendly IPCs.

Changeling aren't team antags and moreover, when do they have friendly IPCs, ever? Tatorcling and teaming up with another traitor? In that case see 'traitors aren't a team antag'

>Its also weird that, say, nuke ops aren't immune to each others' bullets. Why should cultists be immune to each others' EMP?

Bullets aren't a screen-covering, indiscriminate AoE weapon, though. It's a lot easier to avoid friendly fire with bullets.

That just leaves wizards as a valid comparison. But wizards have a lot of tools to chose from. They can simple not pick the emp spell and instead spend their points on another, more useful spell if they're an IPC. Cult doesn't have limited resources to spend on a large toolbox to chose from. Cult is expected to use every tool at their disposal. Becoming completely unable to safely use one of their basic tools because someone converted an IPC would make for rather bad gameplay, imo.

**Comment by Cocacolagua** [comment]
To answer to the wizard point, they can use the magic mirror on their wizard den to change their species to something else, so they don’t have to suffer the IPC + disable tech amazing synergy

**Comment by ExusA** [comment]
Being a team antag where most people not knowing how to play it well and not knowing the mechanics is seen by a few people as the gamemode's biggest weakness, it makes sense that this particular window for friendly fire doesn't need to exist. It's hard enough to do cult well even with full knowledge of the mechanics, and the fact that it's so easy to directly kill some of your teammates out of not knowing the mechanics doesn't help that issue. Plenty of new players WILL be culted on your average cult round, and they might not know what EMPs do to IPCs.

**Comment by Spacemanspark** [comment]
Why haven't we merged steelslayers PR yet? I've been asking this for _almost an entire year at this point, and it would really help alleviate some of these issues._

**Comment by Pennwick** [comment]
#12628 just so its there.
I feel like one of these two should happen. As it stands I'm not so much bothered by the inability to use the talisman but the rune. Even as a human cultist I never feel like I can justify using it because I might be aware of an IPC cultist but not their location or possibly not even know that we have one if their name is significantly humanish. And as dramatic and fun EMPing the whole station might be before a summon I don't want to take a teammate out of the round.

**Comment by kugamo** [comment]
> 
> 
> Its weird that, of all the exceptions, cult would be the only situation where an EMP doesn't hurt IPCs.
> Its also weird that, say, nuke ops aren't immune to each others' bullets. Why should cultists be immune to each others' EMP?

Besides all the good points that others have made in response to this, and the all are good points. I would also say that nuke ops don't have an evil godly being watching over them and supplying them their bullets/power, cultists do. Why would a godly being knowingly and willingly allow his instruments to use his power to cripple/kill eachother? I don't think it would be weird that a godly being would protect or exclude his followers from the affects of his powers.

If a complete cult overhaul PR was merged and the EMP talisman and rune remained, then this problem hindering the use of them would remain as well.

**Comment by Denthamos** [comment]
This change overall feels a bit snowflaky needing so much code changing done just for a single race on a single game mode.  As echo'd before the better change will be the eventual merge of the IPC balance changes which adjusts how IPC's take damage from EMP's so it would apply to all game modes in which an IPC can be effected by an EMP instead of this bandaid on a single one.

**Comment by Fox-McCloud** [comment]
This is the very definition of snowflake code; it snowflakes a cult spell functionality into `empulse`. This doesn't pass muster, to say the least--this absolutely has to be OOP.

**Comment by kugamo** [comment]
> 
> 
> This change overall feels a bit snowflaky needing so much code changing done just for a single race on a single game mode. As echo'd before the better change will be the eventual merge of the IPC balance changes which adjusts how IPC's take damage from EMP's so it would apply to all game modes in which an IPC can be effected by an EMP instead of this bandaid on a single one.

Well i believe, considering that the way it interacts with IPCs in this one gamemode greatly changes how those in it play, its valid. Even if how IPCs take damage from EMPs is altered it still wouldnt make sense that they could be FFd or lore-wise that the evil godly being granting them their powers would allow that. 

> 
> 
> This is the very definition of snowflake code; it snowflakes a cult spell functionality into `empulse`. This doesn't pass muster, to say the least--this absolutely has to be OOP.

I made some code changes, and made a new proc for cult empulses, hopefully thats more OOP. I am not an experienced coder, so I would appreciate some constructive feedback on my code. Let me know if this change is not enough or give me pointers on how I should approach implementing something like this if how I am doing it is not appropriate. I would like to make it the best code I can.

**Comment by datlo** [comment]
Just FYI this is already handled in the blood cult PR that I recently reopened.

Once blood cult is merged, cultists will be immune to the EMP spells of fellow cultists. (But not their own EMP, this is described pretty explicitely and you get a warning if you have robotic organs.)

**Comment by variableundefined** [comment]
This is a balance, not a fix PR

Objecting for the following reasons: 
1. Recently (1 day ago) merged in a PR that let IPC survive one or two accidental or incidental EMP (provided they're not being immediately attacked) - so the EMP isn't as deadly anymore.
2. Friendly fire are a part of the game - if you use EMP or ion carbine or ion rifle or what have you, accidentally messing up your own borg / IPC / mech are part of the equation. Why shouldn't this apply to cultist either?
3. Code-wise the implementation is snowflakey, requiring cult to uses an entirely new proc for the purpose of avoiding friendly fire.

Reason 1 isn't exactly fair as this PR was made before the IPC balance was merged. 

Since there's two maintainer objections on both code and design ground, I'm closing this PR.

**Pull request closed without merging**